tests/test_csv_readers.py

```diff
@@ -0,0 +1,857 @@
+        },
+    )
+
+    # Check values at a single coordinate
+    coord = {"region": "R1", "timeslice": 1, "commodity": "heat"}
+    assert data.sel(**coord).item() == 0.071
+
+
+def test_read_macro_drivers(correlation_model_path):
+    from muse.readers.csv import read_macro_drivers
+
+    data = read_macro_drivers(
+        correlation_model_path / "residential_presets" / "Macrodrivers.csv"
+    )
+
+    # Check properties of the dataset
+    assert_schema(
+        data,
+        DatasetSchema(
+            dims={"region", "year"},
+            coords={
+                "region": CoordinateSchema(("region",), dtype="<U2"),
+                "year": CoordinateSchema(("year",), dtype="int64"),
+            },
+            data_vars={
+                "gdp": "int64",
+                "population": "int64",
+            },
+        ),
+    )
+
+    # Check coordinate values
+    assert_coordinate_values(
+        data,
+        {
+            "region": ["R1"],
+            "year": list(range(2010, 2111)),
+        },
+    )
+
+    # Check values at a single coordinate
+    coord = {"year": 2010, "region": "R1"}
+    expected = {
+        "gdp": 1206919,
+        "population": 80004200,
+    }
+    assert_single_coordinate(data, coord, expected)
+
+
+def test_read_regression_parameters(correlation_model_path):
+    from muse.readers.csv import read_regression_parameters
+
+    data = read_regression_parameters(
+        correlation_model_path / "residential_presets" / "regressionparameters.csv"
+    )
+
+    # Check properties of the dataset
+    assert_schema(
+        data,
+        DatasetSchema(
+            dims={"sector", "region", "commodity"},
+            coords={
+                "sector": CoordinateSchema(("sector",), dtype="<U11"),
+                "region": CoordinateSchema(("region",), dtype="object"),
+                "commodity": CoordinateSchema(("commodity",), dtype="object"),
+            },
+            data_vars={
+                "GDPexp": "float64",
+                "constant": "float64",
+                "GDPscaleLess": "float64",
+                "GDPscaleGreater": "float64",
+                "function_type": "<U16",
+            },
+        ),
+    )
+
+    # Check coordinate values
+    assert_coordinate_values(
+        data,
+        {
+            "sector": ["residential"],
+            "region": ["R1"],
+            "commodity": ["electricity", "gas", "heat", "CO2f"],
+        },
+    )
+
+    # Check function type
+    assert data.function_type.sel(sector="residential").item() == "logistic-sigmoid"
+
+    # Check values at a single coordinate
+    coord = {"sector": "residential", "region": "R1", "commodity": "heat"}
+    expected = {
+        "GDPexp": 0.0994,
+        "constant": 1.01039e-05,
+        "GDPscaleLess": 753.1068725,
+        "GDPscaleGreater": 672.9316672,
+    }
+    assert_single_coordinate(data, coord, expected)
+
+
+def generate_dataarray_schema(data: xr.DataArray) -> DataArraySchema:
```

Comment:
I would make this and the next one class methods for the `DataArraySchema` and the `DatasetSchema`, so they can be used like:

```python
schema = DataArraySchema.from_da(data)
schema = DatasetSchema.from_ds(data)
```